### PR TITLE
externalresource: add Broker and Manager

### PR DESCRIPTION
**Pull request by liuzix**
ref #240 
- Implement basic `Broker` and `Manager` in `pkg/externalresource`.

### CODE REVIEW

**Review by lance6716 — COMMENTED** [review]
almost LGTM, I'll review later

File: pkg/externalresource/broker/broker.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	}
+}
+
+func (i *Impl) OpenStorage(
+	ctx context.Context,
+	workerID resourcemeta.WorkerID,
+	jobID resourcemeta.JobID,
+	resourcePath resourcemeta.ResourceID,
+) (Handle, error) {
+	if strings.HasPrefix(resourcePath, "/"+string(resourcemeta.ResourceTypeLocalFile)+"/") {
```
[the remark this answers is not in the document]
Comment (liuzix) [reply pkg/externalresource/broker/broker.go:53]: In this design, a path similar to `/local/{uuid}` is used.

File: pkg/externalresource/resourcemeta/accessor.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	resp, err := m.client.Get(ctx, keyPrefix, metaclient.WithPrefix())
+	if err != nil {
+		return nil, err
+	}
+	timeCost := time.Since(startTime)
+	log.L().Info("Scanned all resources",
+		zap.Int("count", len(resp.Kvs)),
+		zap.Duration("duration", timeCost))
+
+	for _, kv := range resp.Kvs {
```
Comment (amyangfei) [inline pkg/externalresource/resourcemeta/accessor.go:143]: Can leave TODO, that the code can be optimized when metastore supports querying with condition.

File: pkg/externalresource/broker/storage_handle.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		jobID:    jobID,
+		workerID: workerID,
+	}, nil
+}
+
+func getPathSuffix(prefix resourcemeta.ResourceType, path resourcemeta.ResourceID) (string, error) {
+	if !strings.HasPrefix(path, "/"+string(prefix)+"/") {
+		return "", derror.ErrUnexpectedResourcePath.GenWithStackByArgs(path)
+	}
+	return strings.TrimSuffix(path, "/"+string(prefix)+"/"), nil
```
Comment (amyangfei) [inline pkg/externalresource/broker/storage_handle.go:103]: s/TrimSuffix/TrimPrefix ?

**Review by amyangfei — APPROVED** [review]

**Pull request merged**